software/spmd/asm_dmem_test/main.S

```diff
@@ -5,10 +5,6 @@
 #include "bsg_manycore_asm.h"
 #define N 1024
 
-// allocate 4KB in DMEM
-.data
-  dmem_arr: .space (N*4)
-
```

Comment:
isn't the test still allocating 4KB in DMEM?

Reply:
On a related note, dmem_arr was a variable that could be referenced. Was there a reason it was replaced by a 0 offset? 

It seems nice to keep things like this. If we ever move the location of DMEM, or change where DMEM allocates data pushing the dereferencing logic into the compiler would save us a lot of work -- like all the changes in the .S files here.

Reply:
The programmer should know that DMEM starts from 0. It's not necessary to create .data